tests/test_event.py

```diff
@@ -0,0 +1,125 @@
+import datetime
+
+import pytest
+
+import toadr3
+
+UTC = datetime.timezone.utc
+
+
+def create_event() -> dict:
+    event = {
+        "id": "37",
+        "createdDateTime": "2024-08-15T08:52:55.578Z",
+        "modificationDateTime": "2024-08-15T08:53:41.127Z",
+        "objectType": "EVENT",
+        "programID": "69",
+        "eventName": "powerLimit",
+        "targets": [
+            {"type": "RESOURCE_NAME", "values": [1211]},
+            {"type": "ORGANIZATION_ID", "values": ["1337"]},
+        ],
+        "reportDescriptors": [
+            {
+                "payloadType": "POWER_LIMIT_ACKNOWLEDGEMENT",
+                "aggregate": False,
+                "startInterval": 0,
+                "numIntervals": -1,
+                "historical": False,
+                "frequency": -1,
+                "repeat": 1,
+                "targets": [],
+            }
+        ],
+        "payloadDescriptors": [
+            {
+                "payloadType": "CONSUMPTION_POWER_LIMIT",
+                "units": "KW",
+                "objectType": "EVENT_PAYLOAD_DESCRIPTOR",
+            }
+        ],
+        "intervalPeriod": {
+            "start": "2024-08-15T10:00:00.000Z",
+            "duration": "PT15M",
+            "randomizeStart": "PT0S",
+        },
+        "intervals": [
+            {"id": 0, "payloads": [{"type": "CONSUMPTION_POWER_LIMIT", "values": [1000]}]}
+        ],
+    }
+    return event
+
+
+def test_event():
+    event = toadr3.Event(create_event())
+
+    assert event.id == "37"
+    assert event.created == datetime.datetime(2024, 8, 15, 8, 52, 55, 578000, tzinfo=UTC)
+    assert event.modified == datetime.datetime(2024, 8, 15, 8, 53, 41, 127000, tzinfo=UTC)
+    assert event.program_id == "69"
+    assert event.event_name == "powerLimit"
+    assert event.priority is None
+
+    assert len(event.targets) == 2
+    assert event.targets == {"RESOURCE_NAME": [1211], "ORGANIZATION_ID": ["1337"]}
```

Comment:
No, however, I am taking a chance until I find better documentation or we fail horribly. 

One solution would be to append the arrays if they are identical, at the moment I raise an exception if there are duplicates.

Semantically I am unsure what it would mean to have duplicate `targets` or `payloads`.

Reply:
I will document this assumption in the parsing function.